src/Object/Contact.php

```diff
@@ -0,0 +1,830 @@
+<?php
+
+/**
+ * @file src/Object/Contact.php
+ */
+
+namespace Friendica\Object;
```

Comment:
Why is it in this namespace and not in "Friendica\Model"?

Reply:
In my view, Models should only comprise database function related to a specific table. Objects are classes that don't fit this description, like `Item` and `Conversation`. My intention is to have a separate `Contact` Model class which holds the database function related to the `contact` table the way `Model\GlobalContact` holds the database function related to the `gcontact` table.

And yes, we can have a `Model\Contact` and an `Object\Contact` invoked in the same file, the `use` statements would look like that: 
````php
use Friendica\Model\Contact as ContactModel;
use Friendica\Object\Contact as ContactObject;
````

Reply:
@MrPetovan 
I completely agree with you. But I think some things are a bit inconsistent (or I don't understand the difference).
E.g. Why is `User::remove()` a `Model` and `Contact::remove()` an `Object`? From my point of view there are many methods in `Friendica\Object\Contact` which deal with database functions related to the `contact` table (`markForArchival()`, `getDetailsByAddr()`, `getUngroupedList()`, `getUngroupedList()`).
Can you explain the difference?

Reply:
The difference is that the point of this PR was to remove ˋinclude/Contact.php`, and while adding ˋModel\User` wasn’t a big deal, adding ˋModel\Contact` would have made the PR significantly more complicated because of the namespace conflict.

However, it is planned as a future improvement because many methods currently in `Object\Contact` indeed belong to ˋModel\Contact` and I’m as little satisfied as you with this current state.

Reply:
Although it is technically possible, I would like to avoid having ```Model\Contact``` and ```Object\Contact```, because I think this would be confusing. Especially when you are searching for the usage of a method or class it would be best if it was named identically at every place in the code.

Reply:
Gotcha, I'll keep that in mind. I was just following the other two non-Model classes naming (`Object\Item` and `Object\Conversation`) but I can come up with something else for all of them.